app.rb

```diff
@@ -0,0 +1,177 @@
+require_relative 'classroom'
+require_relative 'person'
+require_relative 'student'
+require_relative 'teacher'
+require_relative 'book'
+require_relative 'rental'
+
+class App
+  def initialize
+    @people = []
+    @books = []
+    @rentals = []
+  end
+
+  def display_menu
+    puts 'Welcome to the School Library App'
+    puts 'Please choose an option by entering a number:'
+    puts '1. List all books'
+    puts '2. List all people'
+    puts '3. Create a person'
+    puts '4. Create a book'
+    puts '5. Create a rental'
+    puts '6. List all rentals for a given person id'
+    puts '7. Exit'
+  end
+
+  def list_books
+    puts 'List of Books:'
+    @books.each { |book| puts "#{book.title} by #{book.author}" }
+  end
+
+  def list_people
+    puts 'List of People:'
+    @people.each do |person|
+      if person.is_a?(Teacher)
+        puts "[Teacher] Name: #{person.name} ID: #{person.id}: Age: #{person.age}"
+      elsif person.is_a?(Student)
+        puts "[Student] Name: #{person.name} ID: #{person.id}: Age: #{person.age}"
+      else
+        puts "Person Name: #{person.name} ID: #{person.id}: Age: #{person.age}"
+      end
+    end
+  end
+
+  def create_person
+    puts 'Create a Person'
+    puts 'Do you want to create a student (1) or a teacher (2)? [Input the number]:'
```

Comment:
Great job team 🤝
- As stated in the project requirements, we are supposed to respect the `Single Responsibility Rule`. Kindly ensure all requests for input and input validation have been placed in the main.rb file and/or classes/methods whose responsibilities are to request or validate input. Kindly implement this across all methods that do not satisfy the Single Responsibility Rule.

Reply:
Done, thank you very much for your review, this is my last extra review, I hope I can approve 💪😃